### PR TITLE
Add test of ASM bug 317792

### PR DESCRIPTION
**Pull request by Godin**

### CODE REVIEW

File: org.jacoco.core.test/src/org/jacoco/core/test/validation/ResizeInstructionsTest.java [hunk 1]
```diff
@@ -48,6 +49,48 @@ public void teardown() {
...
+				addCauseOfResizeInstructions(mv);
+				mv.visitMaxs(2, 1);
+				mv.visitEnd();
+				super.visitEnd();
+			}
+		}, 0);
+		final byte[] bytes = instrumenter.instrument(cw.toByteArray(), "");
+
+		final TargetLoader targetLoader = new TargetLoader();
+		final Class outer = targetLoader.add(ResizeInstructionsTest.class,
```
Comment (marchof) [inline org.jacoco.core.test/src/org/jacoco/core/test/validation/ResizeInstructionsTest.java]: @Godin Raw `Class` types give a compiler warning, `Class<?>` fixes the issue.

File: org.jacoco.core.test/src/org/jacoco/core/test/validation/ResizeInstructionsTest.java [hunk 2]
```diff
@@ -48,6 +49,48 @@ public void teardown() {
...
+				TargetLoader.getClassDataAsBytes(ResizeInstructionsTest.class));
+		final Class inner = targetLoader.add(Inner.class, bytes);
```
Comment (marchof) [inline org.jacoco.core.test/src/org/jacoco/core/test/validation/ResizeInstructionsTest.java]: @Godin Same here: Raw `Class` types give a compiler warning, `Class<?>` fixes the issue.

File: org.jacoco.core.test/src/org/jacoco/core/test/validation/ResizeInstructionsTest.java [hunk 3]
```diff
@@ -48,6 +49,48 @@ public void teardown() {
...
+		assertNull(inner.getEnclosingClass());
+		assertNotSame(outer, inner.getDeclaringClass());
+		assertNull(inner.getDeclaringClass());
+	}
+
+	/**
+	 * Test of ASM bug
+	 * <a href= "https://gitlab.ow2.org/asm/asm/issues/317630">#317630</a> that
+	 * caused {@code java.lang.ClassNotFoundException}.
+	 */
```
Comment (marchof) [inline org.jacoco.core.test/src/org/jacoco/core/test/validation/ResizeInstructionsTest.java]: @Godin For consistency this test should also have a descriptive method name.

File: org.jacoco.core.test/src/org/jacoco/core/test/validation/ResizeInstructionsTest.java (hunk 1 above)
Comment (Godin) [reply org.jacoco.core.test/src/org/jacoco/core/test/validation/ResizeInstructionsTest.java]: @marchof fixed

File: org.jacoco.core.test/src/org/jacoco/core/test/validation/ResizeInstructionsTest.java (hunk 2 above)
Comment (Godin) [reply org.jacoco.core.test/src/org/jacoco/core/test/validation/ResizeInstructionsTest.java]: @marchof fixed

File: org.jacoco.core.test/src/org/jacoco/core/test/validation/ResizeInstructionsTest.java (hunk 3 above)
Comment (Godin) [reply org.jacoco.core.test/src/org/jacoco/core/test/validation/ResizeInstructionsTest.java]: @marchof renamed

**Pull request merged**